24_Read処理のServiceとController部分を実装の課題

javachead opened the pull request:
![23](https://github.com/user-attachments/assets/04172e61-f065-4e9d-b3f1-a51503bcf029)
![23-2](https://github.com/user-attachments/assets/8a7f6260-16d7-42de-ad1e-b8dc20c2f1e1)

23の課題を提出します

Commit:
23の課題

Commit:
Apache3追加

Commit:
Apache3追加

Commit:
学生リストの全取得

mmanatumi committed:
23の課題

TomooAkamatsu reviewed (commented):
余計な差分が多いとレビュー時のノイズとなるので注意しましょう
例えば、`/lib`ディレクトリは完全に不要かと！

javachead commented on src/main/java/raisetech/student/management/StudentRepository.java:
```diff
@@ -1,32 +1,15 @@
+	List<Student> search();
+
+	@Select("SELECT * FROM student_courses")
+	List<StudentCourse> searchcourselist();
```
List<StudentCourse> searchCourseList();に修正いたします。
ご確認いただき誠にありがとうございます

javachead commented:
> 余計な差分が多いとレビュー時のノイズとなるので注意しましょう 例えば、`/lib`ディレクトリは完全に不要かと！
削除いたしますありがとうございます

javachead commented:
> ![23](https://private-user-images.githubusercontent.com/182053859/382785814-04172e61-f065-4e9d-b3f1-a51503bcf029.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4MTQzODksIm5iZiI6MTczMDgxNDA4OSwicGF0aCI6Ii8xODIwNTM4NTkvMzgyNzg1ODE0LTA0MTcyZTYxLWYwNjUtNGU5ZC1iM2YxLWE1MTUwM2JjZjAyOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNVQxMzQxMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Yjk4ZDg2Y2Y5ZmMwMDdiMjI3ZGNjMzZlMWNjOGJmYmU2MjExMWNkZTg3YjMwOTkzMjU2ZjkwNTQ5MDcxZTY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.n7J3Si4---rl6rkAZ4bT616gYQ4yl9VdMBFHA4fTpNo) ![23-2](https://private-user-images.githubusercontent.com/182053859/382785855-8a7f6260-16d7-42de-ad1e-b8dc20c2f1e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4MTQzODksIm5iZiI6MTczMDgxNDA4OSwicGF0aCI6Ii8xODIwNTM4NTkvMzgyNzg1ODU1LThhN2Y2MjYwLTE2ZDctNDJkZS1hZDFlLWI4ZGMyMGMyZjFlMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNVQxMzQxMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02OTUwYmFlMTQ5NTFhOTdhNzlhMGUwZDg3MWYzZjZmMTkwNDM1MWYwYmQ5NTA1MjI3NWIyZTU1OGJhM2Q2ZjhiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-u9vgNlgmitcxKcJd9L63-75fywhRBk38ksuS8aayzo)
> 
> 23の課題を提出します

課題
①StudentCourseクラスの作成
②Repositoryに全件検索のメソッドを追加
③Applicationに全件検索を表示させるメソッドを追加

mmanatumi committed:
23課題の修正
①コミットメッセージの修正
②ノイズ軽減のため/libディレクトリの削除
③List<StudentCourse> searchCourseList();キャメルケースへの修正
④24の課題途中の為、一部パッケージ化し振り分けています

mmanatumi committed:
23課題の修正
①フォーマットのずれを修正

Commit:
23課題の修正
①コミットメッセージの修正
②ノイズ軽減のため/libディレクトリの削除
③List<StudentCourse> searchCourseList();キャメルケースへの修正
④24の課題途中の為、一部パッケージ化し振り分けています

TomooAkamatsu reviewed (commented):
必要であればgitignoreへの追記も行っておきましょう！

mmanatumi committed:
フォーマットの修正

Commit:
特定条件でのデータ抽出

Commit:
フォーマットの修正＆gitignoreへの追記

mmanatumi committed:
Resolved merge conflicts

javachead committed:
Delete .gitignore

インデント変更のみでコードの修正は前回と比べてなかった為、ノイズ削減で削除しました。

javachead committed:
Delete build.gradle

インデント変更のみでコードの修正は前回と比べなかった為、ノイズ削減で削除しました。

javachead committed:
Delete build.groovy

インデント変更のみでコードの修正は前回と比べなかった為、ノイズ削減で削除しました。

javachead committed:
Delete gradle/wrapper/gradle-wrapper.properties

インデント変更のみでコードの修正はなかった為ノイズ削減の為、削除しました。

javachead committed:
Delete gradlew.bat

インデント変更のみでコードの修正は前回と比べなかった為、ノイズ削減で削除しました。

javachead committed:
Delete src/main/resources/application.properties

インデント変更のみでコードの修正は前回と比べなかった為、ノイズ削減で削除しました。

javachead committed:
Delete src/test/java/raisetech/student/management/ApplicationTests.java

インデント変更のみでコードの修正は前回と比べなかった為、ノイズ削減で削除しました。

javachead reviewed (commented):
お世話になっております。
講義24の課題を提出いたします。
**修正をしたファイルだけPRしなおしました。**
ご確認よろしくお願いします　※23の課題の継続でPRしています。

①パッケージ作成＋データの振り分け
②Javadocの追加
③サービスの作成
④コントローラーの作成
⑤上記作成後、以下課題

【1】年齢が30代の人のみを抽出し、そのリストをコントローラーに返す
【2】絞り込み検索で「JAVAコース」のコース情報のみを抽出し、そのリストをコントローラーに返す 

**MysqlのDB**
![-3](https://github.com/user-attachments/assets/f9307158-b966-493c-b6a3-2d9d9879217c)
![24-5](https://github.com/user-attachments/assets/a02a82f8-c918-4541-a5ef-330c89c1ec5d)

**結果**
![24](https://github.com/user-attachments/assets/1ef75ec2-d8b0-4af3-82f6-de4c6dc44449)
![24-2](https://github.com/user-attachments/assets/b026af2d-dac3-4437-8290-01bf74e879bc)

javachead commented:
https://github.com/javachead/ServiceController/pull/1へ移動
